src/backend/distributed/commands/utility_hook.c

```diff
@@ -1131,8 +1189,16 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 			 * will already be in the hash table, hence we won't be holding any snapshots.
 			 */
 			WarmUpConnParamsHash();
+
+			/* similar to pg >= 14 case, pop the active snapshot if exists */
+			if (ActiveSnapshotSet())
+			{
+				PopActiveSnapshot();
+			}
```

Comment:
there's a lot of repetition, can we reduce it to something like:
```c
#if PG_VERSION_NUM < 140000
WarmUpConnParamsHash();
#endif

if (ActiveSnapshotSet())
{
    PopActiveSnapshot();
}

CommitTransactionCommand();
StartTransactionCommand();

#if PG_VERSION_NUM >= 140000
set_indexsafe_procflags();
#endif
```

Reply:
That makes a lot sense, I should've already done that before, thank you